### ISSUE TITLE
How to correctly load local assets like audio files

### ISSUE THREAD

**Issue by mathildebuenerd** [parcel-bundler/parcel#1911]
# ❔ Question

How to correctly load local assets like audio files? I try to load a simple sound like that:

`let sound = new Audio("../sounds/happy/happy-0.mp3");`

But I keep getting a `GET http://localhost:1234/sounds/happy/happy-0.mp3 404 (Not Found)` error.

## 🔦 Context

I am building a kind of web radio using mp3 files. I want to preload multiple files in order to be able to play them after. I use Typescript and parcel. I use this command to start the server: `parcel --no-cache index.html --public-url /`
My folder is structured like that: My code is in loadSoundFiles.ts

<img src="https://image.ibb.co/du5fHe/structure.png" alt="structure">

## 💻 Code Sample

`let sound = new Audio("../sounds/happy/happy-0.mp3");`

I am using variables for generating the url. When I `console.log` the url I get `../sounds/happy/happy-0.mp3`

The actual code is:

```typescript
 audioFiles = [];
    loaded = 0;

    preload(numberOfSounds: object): void {
        // iterates through all the files
        for (const emotion in numberOfSounds) {
            for (let i=0; i<numberOfSounds[emotion]; i++) {
                const url = `./../sounds/${emotion}/${emotion}-${i}.mp3`;
                console.log(url); // ../sounds/happy/happy-0.mp3
                this.audioFiles[this.loaded] = new Audio(url);
                this.loaded++;
            }
        }
    }
```

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 1.9.7
| Node             | /
| npm/Yarn         | 5.6.0
| Operating System | Windows 10

**Comment by DeMoorJasper** [comment]
Parcel only loads paths from requires, imports and workers.
Therefore the easiest fix is just adding a require to the code to resolve it into a url and copy the file into dist.

```Javascript
let audioUrl = require('../sounds/happy/happy-0.mp3');
let sound = new Audio(audioUrl);
```

**Comment by mathildebuenerd** [comment]
Thanks! but with this I get a `Uncaught Error: Cannot find module '../sounds/happy/happy-0.mp3'`

I put the sounds folder in dist. So now the relative path should be `sounds/happy/happy-0.mp3` ? The path is relative to the `index.html` that is in the `dist` folder? I tried both but I still get this error.

**Comment by DeMoorJasper** [comment]
You could use `require('../sounds/*/*.mp3');` this would return an object with mapping from original to hashed bundle.

**Comment by bozdoz** [comment]
I'm having the same problem.  Is there a simple asset plugin that would solve this?  I know with webpack you can just use a `file-loader`.

**Comment by mischnic** [comment]
@bozdoz Have you tried using a glob as suggested?
The code from https://github.com/parcel-bundler/parcel/issues/1911#issuecomment-414935461 is very hard to parse statically to determine which files to bundle.

**Comment by bozdoz** [comment]
Tried a glob as suggested; still could not load audio.  I just switched to webpack for my project.

**Comment by mischnic** [comment]
Mhh, running this packs all mp3 files in `sounds` and clicking in the browser window will then play all of them at once.

```js
const files = Object.values(require('./sounds/*.mp3'));
const audios = files.map(v => new Audio(v));
document.body.onclick = () => {
	audios.forEach(v => v.play());
};
```

Could you please be more specific about "still could not load audio"?

**Comment by mischnic** [comment]
Closing because of inactivity

**Comment by Offirmo** [comment]
I tried with a glob:
- works only with parcel (not compatible with node)
- works only with a true path, not with `require('some-module/**/*.mp3')`

**Comment by ELI7VH** [comment]
I am getting the error:
```
Error: No transformers found for /.../src/audio/eli7vh-tezos-till-i-bezos-final.wav.
```

**Comment by jeremymec** [comment]
Also getting the error about No transformers found

**Comment by mischnic** [comment]
With Parcel 2, you need to prefix the import with `url:`: https://v2.parceljs.org/getting-started/migration/#importing-non-code-assets-from-javascript

**Comment by karltaylor** [comment]
> With Parcel 2, you need to prefix the import with `url:`: https://v2.parceljs.org/getting-started/migration/#importing-non-code-assets-from-javascript

If this isn't immediately clear, (like it was for me pre-coffee)

Change this;

```ts
const track = require("../sounds/track3.mp3");
```

To this

```ts
const track = require("url:../sounds/track3.mp3");
```

And it will load the correct uri

**Comment by hemantvardani** [comment]
My issue got solved when, I started using the path as ('assets/static/audio.mp3'), instead of ('./../static/audio.mp3') i.e. complete path.
